constraint-solver/src/solver.rs

```diff
@@ -156,6 +171,17 @@ impl<T: FieldElement, V: Ord + Clone + Hash + Eq + Display + Debug> Solver<T, V>
         !equivalences.is_empty()
     }
 
+    fn exhaustive_search(&mut self) -> Result<bool, Error> {
```

Comment:
Document? I would actually move the long comment from above here.

Reply:
Agree, but I think the same applies to the other functions called in `loop_until_no_progress` :)